test/integration_test.py

```diff
@@ -147,42 +147,35 @@ def tearDown(self):
     def test(self):
         if config.deployment_stage != 'prod':
             try:
-                self._test_indexing()
-            except Exception as e:
-                log.error('Indexing while running integration test failed,'
-                          'cleaning up ...', exc_info=e)
+                log.info('Starting test using test name, %s ...', self.test_name)
+                azul_client = self.azul_client
+                self.test_notifications, self.expected_fqids = self._test_notifications(test_name=self.test_name,
+                                                                                        test_uuid=self.test_uuid,
+                                                                                        max_bundles=self.max_bundles)
+                self.num_bundles = len(self.expected_fqids)
+                self.queues.wait_for_queue_level(empty=True,
+                                                 num_expected_bundles=self.num_bundles,
+                                                 min_timeout=self.min_timeout)
+                azul_client._index(self.test_notifications)
+                # Index some bundles again to test that we handle duplicate additions.
+                # Note: random.choices() may pick the same element multiple times so
+                # some notifications will end up being sent three or more times.
+                azul_client._index(random.choices(self.test_notifications, k=len(self.test_notifications) // 2))
+                self._check_bundles_are_indexed(self.test_name, 'files')
+            except Exception:
                 self._delete_bundles(self.test_notifications)
+                raise Exception('Exception while indexing. Cleaning up ...')
```

Comment:
When you raise the exception, you've already cleaned up so the exception message makes no sense. Besides, the original exception is swallowed. Log the excepttion first, announce the cleanup, then reraise the original exception.